### DISCUSSION TITLE
How to write out arbitrary strings as 2D text annotations to replace "cycle" and "time" in the main Visit viewer window

### DISCUSSION THREAD

**Discussion by nDimensionalSpace** [LLNL/Silo#281] [Q&A] [answered]
Subject line says it all . . .  More context given here:

https://github.com/visit-dav/visit/issues/5242#issuecomment-1043345250
https://github.com/visit-dav/visit/issues/5242#issuecomment-1043632068

Thanks.

**Comment by markcmiller86** [comment, marked as answer]
Hey @nDimensionalSpace ... hmm. Not totally sure by what you mean by 2D here? Can you elaborate? This is via Silo's Fortran interface, right? Is the problem with just dealing with the string length (or shape) in the `DBCreate()` call?

**Reply by nDimensionalSpace** [reply]
No, we use the C interface directly.

Oh, when you said "... in the meantime, you can adjust your DBCreate() to pass the needed string as the char const *fileinfo ...", I took that to mean that the steady solution would be different.  If that is all that is needed, I should be good.

**Reply by nDimensionalSpace** [reply]
@markcmiller86 Heh, so, maybe not surprisingly, we were already using "fileinfo" for something else, but nothing of import.  Indeed, it shows up in the "File Information" dialog, but I will note that I don't see any way to get it on the screen in 3.3.1.  I guess that there is some kind of control I am missing . . .

**Reply by markcmiller86** [reply]
Have a look at [this part of the docs for using `$dbcomment` in an annotation string](https://visit-sphinx-github-user-manual.readthedocs.io/en/develop/using_visit/MakingItPretty/Annotations.html?highlight=%24dbcomment#database-values-in-text-annotations)

**Reply by nDimensionalSpace** [reply]
So that works, but I have one other question: this seems to require somewhat more . . . interaction than I think some users will be comfortable with.  Is there some easy way to be able to specify (via options defined in Silo) that this guy starts in the same location as "cycle", and also disables both "cycle" and "time"?

(When I say "easy" above, I really mean it . . .  If the answer is "no, not easy" or "I don't have the time right now" or "take your pita questions elsewhere", I am completely ok with that.  Like, this is a great improvement, even if it is just for me for the time being. 🙏🙏)

**Reply by markcmiller86** [reply]
Hmmm...what we could possibly consider is something in Silo that informs downstream tools (aka VisIt) what annotations to use. I'd have to think about that a bit because the concept of annotation is a bit nebulus. Sometimes its just some arbitrary string. Sometimes its some metadata *derived* from the database (e.g. time or cycle), etc.

That aside, the questions you ask are all related to how VisIt handles annotations and you can turn off cycle & time with the check-box in for `time` in **Controls->Annotations->General**.

As to this being easy for a user who might not want to deal with all the additional interactions, there are a few options...

- Macros...some python code that you can share with users that goes into `~/.visitrc`
- Sessions and sharing session files
- Saving settings and sharing the settings...this can be harder than it sounds though
